src/components/MyCheckListItems/MyCheckListItems.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  IColumn,
+  IGroup,
+  SelectionMode,
+  ShimmeredDetailsList,
+} from "@fluentui/react";
+import { Button, makeStyles } from "@fluentui/react-components";
+import {
+  ICheckListItem,
+  useOpenChecklistItems,
+  useUpdateCheckListItem,
+} from "api/CheckListItemApi";
+import { IInRequest, useRequests } from "api/RequestApi";
+import { Link } from "react-router-dom";
+import { UserContext } from "providers/UserProvider";
+import { RoleType } from "api/RolesApi";
+import { useContext, useState } from "react";
+import { useError } from "hooks/useError";
+import { ApiError } from "api/InternalErrors";
+
+/** ICheckListItem extended by adding the request info  */
+interface ICheckListItemLookup extends ICheckListItem {
+  request: IInRequest;
+}
+
+/** FluentUI Styling */
+const useStyles = makeStyles({
+  myCheckListItemsHeader: {
+    paddingLeft: "1em",
+    paddingRight: "1em",
+  },
+});
+
+/** This is a component to display the Active CheckListItems that the current user can complete */
+export const MyCheckListItems = () => {
+  /** FluentUI Styling */
+  const classes = useStyles();
+
+  /** The user from UserContext */
+  const user = useContext(UserContext);
+
+  /** Hook to get the CheckListItems that don't have a Completion Date  */
+  const { data: checklistItems } = useOpenChecklistItems();
+
+  /** Hook to update a CheckListItem */
+  const { completeCheckListItem } = useUpdateCheckListItem();
+
+  /** Hook to get the requests */
+  const { data: requests } = useRequests();
+
+  /** Error Handling Hook */
+  const errorObj = useError();
+
+  /** This is a type used to create an object to store whether the group is collapsed */
+  type groupcollapse = { [key in RoleType]: boolean };
+
+  /** The object that tells the ShimmeredDetailList how to display the groups */
+  let groups: { all: IGroup[]; currentItem?: IGroup } | undefined;
+
+  // State to hold whether or not the group is collapsed or not
+  // This way when the screen is re-rendered based on the items updating (added/removed)
+  // It leaves the group in the collapsed/uncollapsed state it was in
+  const [groupCollapsedState, setGroupCollapsedState] = useState<groupcollapse>(
+    // Initialize all RoleTypes as collapsed
+    Object.values(RoleType).reduce((prevVal, curVal) => {
+      prevVal[curVal] = true;
+      return prevVal;
+    }, {} as groupcollapse)
+  );
+
+  /** Listing of requests stored by the Id */
+  let requestLookup: {
+    [k: string]: IInRequest;
+  };
+
+  /** The current user's CheckListItems */
+  let myCheckListItems: ICheckListItemLookup[] | undefined;
+
+  // Ensure we have all the data we need to correctly determine which items to show
+  if (checklistItems && requests && user?.user && user?.roles) {
+    // Create an object where we can quickly reference the Request based on the Id
+    requestLookup = Object.fromEntries(
+      requests.map((request) => [request.Id.toString(), request])
+    );
```

Comment:
I'll take a look.  When I was doing this, I did notice that I don't think my toString() was actually doing anything, because it still looked like a number in Dev tools.